Changelog.md

```diff
@@ -20,6 +20,7 @@ Compiler Features:
  * Yul Optimizer: Try to simplify function names.
  * Yul IR Generator: Report source locations related to unimplemented features.
  * Optimizer: Optimize ``exp`` when base is 0, 1 or 2.
+ * Yul: Specialize checked exponentiation when the base of the ``exp`` is a literal.
```

Comment:
We usually don't put changelog entries for sol-yul codegen. The entry in line 21 is an exception because it was requested by users.